lib/core/src/refund/chain.rs

```diff
@@ -0,0 +1,273 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use boltz_client::{
+    boltz::{BOLTZ_MAINNET_URL_V2, BOLTZ_TESTNET_URL_V2},
+    network::electrum::ElectrumConfig,
+    util::secrets::Preimage,
+    Amount, BtcSwapTx,
+};
+use log::info;
+use tokio::sync::Mutex;
+
+use crate::{
+    chain::{bitcoin::BitcoinChainService, liquid::LiquidChainService},
+    error::{PaymentError, SdkResult},
+    prelude::{ChainSwap, Config, Direction, LiquidNetwork, SwapScriptV2},
+    swapper::BoltzSwapper,
+};
+
+use super::{
+    common::{self, get_lbtc_fee_rate_sats_per_vbyte},
+    RefundHandler,
+};
+
+pub(crate) struct ChainRefundHandler<SwapperType: ?Sized> {
+    config: Config,
+    swapper: Option<Arc<SwapperType>>,
```

Comment:
Why is this an `Option` if it's always set when initializing this in `new`?

Reply:
The idea is it won't be always set, since the refunder shouldn't always interact with a swapper. We can do this by adding an implementation of ChainRefundHandler<()>, which would then set the swapper field to `None`.